test/test_invalid_headers.py

```diff
@@ -365,6 +367,27 @@ def test_push_promise_skip_normalization(self, frame_factory, headers):
         )
         assert c.data_to_send() == pp_frame.serialize()
 
+    @pytest.mark.parametrize('headers', strippable_header_blocks)
+    def test_strippable_headers(self, frame_factory, headers):
+        """
+        Test connection related headers are removed before sending.
+        """
+        c = h2.connection.H2Connection()
+        c.initiate_connection()
+
+        # Clear the data, then try to send headers.
+        c.clear_outbound_data_buffer()
+        c.send_headers(1, headers)
+
+        conn_headers = (
+            'connection', 'proxy-connection', 'keep-alive',
+            'transfer-encoding', 'upgrade'
+        )
+        stripped_headers = [h for h in headers if h[0] not in conn_headers]
+
+        f = frame_factory.build_headers_frame(stripped_headers)
+        assert c.data_to_send() == f.serialize()
+
```

Comment:
So we need more testing here.

In particular, we need to test that these headers are left in place (and therefore rejected) if `normalize_header_blocks` is not turned on. We also need to test that these header blocks fail on receipt unless `validate_inbound_headers` is turned off. Essentially, you need to go back to the following tests and have them also test `stripped_headers` to ensure that `stripped_headers` does the right thing:

- `TestSendingInvalidFrameSequences.test_headers_event_skipping_validation` (should emit the header block without the stripped field but otherwise left intact)
- `TestSendingInvalidFrameSequences.test_push_promise_skipping_validation` (should emit the header block without the stripped field but otherwise left intact)
- `TestSendingInvalidFrameSequences.test_headers_event_skip_normalization`
- `TestSendingInvalidFrameSequences.test_push_promise_skip_normalization`